### PR TITLE
Fix text overflow in branch/tag selector

### LINKED ISSUE #27971: Long name branch/tag display problem in repo home page
### Description

image

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build

### Database

None

### PR DESCRIPTION
**Pull request by SohlRing**
Fixes #27971 

Before: Long branch/tag names overflow the selector button.
Branch/tag name overflows such that only the middle of the name is visible

After: uses `text-overflow: ellipsis;` to not do that.
Branch/tag names truncated by ellipsis in the selector box

### CODE REVIEW

File: templates/repo/branch_dropdown.tmpl [hunk 1]
```diff
@@ -70,8 +70,8 @@
 <div class="js-branch-tag-selector {{if .ContainerClasses}}{{.ContainerClasses}}{{end}}">
 	{{/* show dummy elements before Vue componment is mounted, this code must match the code in BranchTagSelector.vue */}}
 	<div class="ui dropdown custom">
-		<button class="branch-dropdown-button gt-ellipsis ui basic small compact button gt-df gt-m-0">
-			<span class="text gt-df gt-ac gt-mr-2">
+		<button class="branch-dropdown-button ui basic small compact button gt-df gt-m-0">
+			<span class="text gt-df gt-ac gt-mr-2" style="overflow: hidden">
```
Comment (silverwind) [inline templates/repo/branch_dropdown.tmpl]: We could add a `gt-overflow-hidden` to `helpers.css`, better than inline style at least.

Comment (SohlRing) [reply templates/repo/branch_dropdown.tmpl]: I found an existing utility class that worked. I see there's already been discussion about using Tailwind too.

**Comment by SohlRing** [comment]
Welcome back descenders! (see letters g p q)
Capture

**Review by silverwind — APPROVED** [review]

**Review by denyskon — APPROVED** [review]

File: templates/repo/branch_dropdown.tmpl [hunk 2]
```diff
@@ -70,8 +70,8 @@
 <div class="js-branch-tag-selector {{if .ContainerClasses}}{{.ContainerClasses}}{{end}}">
 	{{/* show dummy elements before Vue componment is mounted, this code must match the code in BranchTagSelector.vue */}}
 	<div class="ui dropdown custom">
-		<button class="branch-dropdown-button gt-ellipsis ui basic small compact button gt-df gt-m-0">
-			<span class="text gt-df gt-ac gt-mr-2">
+		<button class="branch-dropdown-button ui basic small compact button gt-df gt-m-0">
+			<span class="text gt-df gt-ac gt-overflow-x-auto gt-as">
```
Comment (wxiaoguang) [inline templates/repo/branch_dropdown.tmpl]: I would suggest to avoid adding more short names. What is an `as` helper .... it's difficult to read.

Comment (SohlRing) [reply templates/repo/branch_dropdown.tmpl]: Agreed. I also think that using `!important` on every helper class is not a good practice, and that reinventing Tailwind is probably not a great idea. But it is what it is, and it's out of scope. There are existing issues that cover this topic like [this one](https://github.com/go-gitea/gitea/issues/27677#issuecomment-1768925072).
https://github.com/go-gitea/gitea/blob/816e46ee7ce4b2649479554a940ecbe1cc505a3d/web_src/css/helpers.css#L5-L16

Comment (silverwind) [reply templates/repo/branch_dropdown.tmpl]: We should use long tailwind-compatible names for new helpers. Important on these helpers will likely remain necessary to overwrite conflicting styles, primarily from Fomantic UI, unfortunately.

File: web_src/css/helpers.css [hunk 3]
```diff
@@ -5,6 +5,7 @@ Gitea's private styles use `g-` prefix.
 .gt-df { display: flex !important; }
 .gt-dib { display: inline-block !important; }
 .gt-ac { align-items: center !important; }
+.gt-items-stretch { align-items: stretch !important; }
```
Comment (silverwind) [inline web_src/css/helpers.css]: Looks good, but could you move it to line 62?

**Comment by yp05327** [comment]
Before:
image
After:
image

We need to check this issue, before merge.

**Comment by yp05327** [comment]
https://github.com/go-gitea/gitea/pull/28010/commits/bb1eedd8787c5360662759015ade75a15ab6ea88 fixed the issue above.

**Review by denyskon — APPROVED** [review]

**Comment by silverwind** [comment]
Seems we forgot this one. I'm not 100% sure if the helper changes still make sense, I will likely check it out with a fresh start.

**Pull request closed without merging**